File: src/SOP8/Utility/dnsData/CommonCode/History.cs
namespace dnsData.CommonCode
{
    public class History
    {
        public class DetectStatus : CodeTypeData
        {
            public override CommonCode.CodeType CodeType
            {
                get { return CommonCode.CodeType.DetectStatus; }
            }

            public const int None = -1;
            public const int Real = (int)CodeType.DetectStatus + 1;          // 실제신호
            public const int Malfunction = (int)CodeType.DetectStatus + 2;   // 오동작
            public const int Test = (int)CodeType.DetectStatus + 3;          // 테스트

            public DetectStatus()
            {
                m_collections.Add(Real);
                m_collections.Add(Malfunction);
                m_collections.Add(Test);
            }
        }

        public class ReactionType : CodeTypeData
        {
            public override CommonCode.CodeType CodeType
            {
                get { return CommonCode.CodeType.ReactionType; }
            }

            public const int None = -1;
            public const int BeginStatus = (int)CodeType.ReactionType + 0;       // 상황시작
            public const int AlarmSignal = (int)CodeType.ReactionType + 1;       // 알람신호
            public const int ClearSignal = (int)CodeType.ReactionType + 2;       // 복구신호
            public const int RunBroadcast = (int)CodeType.ReactionType + 10;     // 방송 실시
            public const int SendSMS = (int)CodeType.ReactionType + 11;          // 문자메시지 발송
            public const int Malfunction = (int)CodeType.ReactionType + 21;      // 오작동 처리
            public const int NotifySignal = (int)CodeType.ReactionType + 22;     // 재난신고
            public const int IgnoreSignal = (int)CodeType.ReactionType + 23;     // 재난탐지신호 무시
            public const int EndStatus = (int)CodeType.ReactionType + 50;        // 상황종료
            public const int ChangeAlarmDepth = (int)CodeType.ReactionType + 62; // 알람 단계 변경
            public const int UserReset = (int)CodeType.ReactionType + 64;        // 사용자 복구
            public const int TimeOut = (int)CodeType.ReactionType + 99;          // 시간초과(알람시간)

            public ReactionType()
            {
                m_collections.Add(BeginStatus);
                m_collections.Add(AlarmSignal);
                m_collections.Add(ClearSignal);
                m_collections.Add(RunBroadcast);
                m_collections.Add(SendSMS);
                m_collections.Add(Malfunction);
                m_collections.Add(NotifySignal);
                m_collections.Add(IgnoreSignal);
                m_collections.Add(EndStatus);
                m_collections.Add(ChangeAlarmDepth);
                m_collections.Add(UserReset);
                m_collections.Add(TimeOut);
            }
        }

        // 문자메시지 또는 이메일의 전송 타입
        public class SendType : CodeTypeData
        {
            public override CommonCode.CodeType CodeType
            {
                get { return CommonCode.CodeType.SendType; }
            }

            public const int None = -1;
            public const int Manual = (int)CodeType.SendType + 0;       // 수동 전송
            public const int AutoFromAlarm = (int)CodeType.SendType + 1;// 알람에 의한 자동 전송(SDMS)
            public const int SOP = (int)CodeType.SendType + 2;          // SOP 실행중 전송

            public SendType()
            {
                m_collections.Add(Manual);
                m_collections.Add(AutoFromAlarm);
                m_collections.Add(SOP);
            }
        }

        public class SopStatus : CodeTypeData
        {
            public override CommonCode.CodeType CodeType
            {
                get { return CommonCode.CodeType.SopStatus; }
            }

            public const int None = -1;
            public const int Ready = (int)CodeType.SopStatus + 0;     // SOP 시작하기 전
            public const int RequestSOP = (int)CodeType.SopStatus + 1;// SOP 실행요청
            public const int RunningSOP = (int)CodeType.SopStatus + 2;// SOP 실행중
            public const int FinishSOP = (int)CodeType.SopStatus + 3; // SOP 종료

            public SopStatus()
            {
                m_collections.Add(Ready);
                m_collections.Add(RequestSOP);
                m_collections.Add(RunningSOP);
                m_collections.Add(FinishSOP);
            }
        }
    }
}

File: src/SOP8/광양/EgovLogin/base-utility/src/main/java/egovframework/base/data/common/code/History.java
package egovframework.base.data.common.code;

import java.util.Arrays;
import egovframework.base.data.common.code.CodeTypeData;
import egovframework.base.data.common.code.CodeType;

public class History {
	public static class DetectStatus extends CodeTypeData {
        public static final int None = -1;
        public static final int Real = CodeType.DetectStatus.getValue() + 1;         // 실제신호
        public static final int Malfunction = CodeType.DetectStatus.getValue() + 2;  // 오동작
        public static final int Test = CodeType.DetectStatus.getValue() + 3;         // 테스트

        public DetectStatus() {
            collections.addAll(Arrays.asList(Real, Malfunction, Test));
        }

        @Override
        public CodeType getCodeType() {
            return CodeType.DetectStatus;
        }
    }

    public static class ReactionType extends CodeTypeData {
        public static final int None = -1;
        public static final int BeginStatus = CodeType.ReactionType.getValue() + 0;
        public static final int AlarmSignal = CodeType.ReactionType.getValue() + 1;
        public static final int ClearSignal = CodeType.ReactionType.getValue() + 2;
        public static final int RunBroadcast = CodeType.ReactionType.getValue() + 10;
        public static final int SendSMS = CodeType.ReactionType.getValue() + 11;
        public static final int Malfunction = CodeType.ReactionType.getValue() + 21;
        public static final int NotifySignal = CodeType.ReactionType.getValue() + 22;
        public static final int IgnoreSignal = CodeType.ReactionType.getValue() + 23;
        public static final int EndStatus = CodeType.ReactionType.getValue() + 50;
        public static final int ChangeAlarmDepth = CodeType.ReactionType.getValue() + 62;
        public static final int UserReset = CodeType.ReactionType.getValue() + 64;
        public static final int TimeOut = CodeType.ReactionType.getValue() + 99;

        public ReactionType() {
            collections.addAll(Arrays.asList(
                BeginStatus, AlarmSignal, ClearSignal, RunBroadcast, SendSMS,
                Malfunction, NotifySignal, IgnoreSignal, EndStatus,
                ChangeAlarmDepth, UserReset, TimeOut
            ));
        }

        @Override
        public CodeType getCodeType() {
            return CodeType.ReactionType;
        }
    }

    public static class SendType extends CodeTypeData {
        public static final int None = -1;
        public static final int Manual = CodeType.SendType.getValue() + 0;
        public static final int AutoFromAlarm = CodeType.SendType.getValue() + 1;
        public static final int SOP = CodeType.SendType.getValue() + 2;

        public SendType() {
            collections.addAll(Arrays.asList(Manual, AutoFromAlarm, SOP));
        }

        @Override
        public CodeType getCodeType() {
            return CodeType.SendType;
        }
    }

    public static class SopStatus extends CodeTypeData {
        public static final int None = -1;
        public static final int Ready = CodeType.SopStatus.getValue() + 0;
        public static final int RequestSOP = CodeType.SopStatus.getValue() + 1;
        public static final int RunningSOP = CodeType.SopStatus.getValue() + 2;
        public static final int FinishSOP = CodeType.SopStatus.getValue() + 3;

        public SopStatus() {
            collections.addAll(Arrays.asList(Ready, RequestSOP, RunningSOP, FinishSOP));
        }

        @Override
        public CodeType getCodeType() {
            return CodeType.SopStatus;
        }
    }
}
